tests: adding k8s GSG to multi node test

aanm opened the pull request:
Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
tests: adding k8s GSG to multi node test

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
tests: fixed wait_for_healthy_k8s_cluster

Fixed helper function to wait until N nodes were ready on the k8s
cluster.

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
tests: add more files to 00-no-sleep.sh

Since commit 82976101cf introduced a diff file containing a "sleep"
command, for the pod specification, it was necessary to add `.yaml` and
`.diff` extensions to the list of ignored extensions in 00-no-sleep.sh.

Signed-off-by: André Martins <andre@cilium.io>

tgraf commented on tests/k8s/multi-node/tests/deployments/gsg/minikube-gsg-l7-fix.diff:
```diff
@@ -0,0 +1,28 @@
+         image: cilium/demo-httpd
+         ports:
+         - containerPort: 80
++      nodeSelector:
```
Are you sure we still need this? L7 policy should work across nodes just fine now.

aanm replied:
Oh, on that case let me test that

aanm replied:
It turns out we need that because the gsg is intended for a single node cluster. For that reason the diff is applied.

tgraf replied:
That doesn't make sense. What part makes it single node only?

tgraf commented on tests/helpers.bash:
```diff
@@ -221,29 +221,43 @@ function wait_for_n_running_pods {
 	local iter=0
 	local found=$(kubectl get cs | grep -v "STATUS" | grep -c "Healthy")
-	until [[ "$found" -eq "$NNODES" ]]; do
+	until [[ "$found" -eq "3" ]]; do
```
Is this guaranteed to always be 3?

aanm replied:
Yes, it's the number of component status

ianvernon commented on tests/helpers.bash:
```diff
@@ -221,29 +221,43 @@ function wait_for_n_running_pods {
 		else
 			kubectl get cs
-			echo -n " [${found}/${NNODES}]"
+			echo "K8S Components ready: [${found}/3]"
```
Shouldn't this still be ${NNODES} instead of '3'?

aanm replied:
No, the NNODES is below, the `kubectl get cs` returns the component status not the nodes status

ianvernon replied:
OK, makes sense.

aanm committed:
tests: removed k8s single node test

- Remove k8s single node test
- Move multi-node tests to previous single node test directory
- Changed relative paths to accommodate the new directory.
- Disabled the gsg runtime test on a single node environment.

Signed-off-by: André Martins <andre@cilium.io>

aalemayhu reviewed (commented):
LGTM.
